source/common/network/listener_impl.cc

```diff
@@ -51,14 +51,15 @@ void ListenerImpl::listenCallback(evconnlistener*, evutil_socket_t fd, sockaddr*
     listener->proxy_protocol_.newConnection(listener->dispatcher_, fd, *listener);
   } else {
     Address::InstanceConstSharedPtr final_remote_address;
-    if (remote_addr->sa_family == AF_INET) {
-      final_remote_address.reset(
-          new Address::Ipv4Instance(reinterpret_cast<sockaddr_in*>(remote_addr)));
+    if (remote_addr->sa_family == AF_UNIX) {
```

Comment:
It's totally fine to do in a follow up (and maybe you are already thinking about this), but I think we need to start thinking about ipv4 vs. ipv6 listener stats. Probably want to do this inside connection handler: https://github.com/lyft/envoy/blob/master/source/server/connection_handler_impl.cc#L134